File: Iteration2/code/CSE3063GRP12/src/features/student/weekly_schedule/WeeklyScheduleView.java
package features.student.weekly_schedule;

import java.util.ArrayList;
import java.util.Map;

import core.enums.CourseDay;
import core.enums.CourseSlot;
import core.models.concretes.Course;

public class WeeklyScheduleView {
    public void showWeeklySchedule(ArrayList<Course> courses) {
        System.out.println("--------------------------------------------------------------------------------------------------------------------");
        System.out.printf("\t| %-60s | %-15s | %-15s |\n", "Course Name", "Course Day", "Course Slot");
        System.out.println("--------------------------------------------------------------------------------------------------------------------");
    
        for (Course course : courses) {
            for (Map.Entry<CourseDay, ArrayList<CourseSlot>> entry : course.getSession().getCourseSessions().entrySet()) {
                String courseName = course.getName();
                CourseDay courseDay = entry.getKey();
    
                for (CourseSlot slot : entry.getValue()) {
                    String courseSlot = slot.getSlot();
                    System.out.printf("\t| %-60s | %-15s | %-15s |\n", courseName, courseDay, courseSlot);
                }
            }
            System.out.println("--------------------------------------------------------------------------------------------------------------------");
        }
    }
    

    public void showErrorMessage(Exception e) {
        System.out.println("Error: " + e.getMessage());
    }

    public void showQuitMessage() {
        System.out.print("Press q to return Main Menu: ");
    }
}

File: Iteration3/code/CSE3063GRP12/src/features/student/weekly_schedule/WeeklyScheduleView.py
from core.enums.CourseSlot import CourseSlot

class WeeklyScheduleView:
    def show_weekly_schedule(self, courses):
        print("-" * 118)
        print(f"\t| {'Course Name':<60} | {'Course Day':<15} | {'Course Slot':<15} |")
        print("-" * 118)

        for course in courses:
            for course_day, course_slots in course.session.get("courseSessions").items():
                course_name = course.name

                for slot in course_slots:
                    # Accessing the value of the enum member
                    member_name = slot.upper()
                    slot_value = getattr(CourseSlot, member_name).value
                    print(f"\t| {course_name:<60} | {course_day:<15} | {slot_value:<15} |")
            
            print("-" * 118)

